### PR TITLE
impl: `eth_getBlockTransactionCountByNumber method`

### LINKED ISSUE #66: feat: Implement `eth_getBlockTransactionCountByNumber` method
## Description

If there is a way to get the block hash from the block number in starknet. Then,

1) Get block hash from block number
2) Call `starknet_getBlockTransactionCount` 
3) Format response.

https://github.com/keep-starknet-strange/rosettanet/blob/main/rpc-call-examples/eth_getBlockTransactionCountByNumber.hurl
## Requirements

- Unit tests have to be implemented. 
- Do not change the existing type structure

## Completion Time

The issue is assumed to be completed in 5 days. If you need more time, please also specify that.

### PR DESCRIPTION
**Pull request by TropicalDog17**
Closes #66

### CODE REVIEW

**Comment by TropicalDog17** [comment]
@ermvrs Hi, could you give some review on this PR. Thanks!

File: src/utils/validations.ts [hunk 1]
```diff
@@ -42,3 +42,16 @@ export function validateBlockHash(blockHash: string): boolean {
 
   return true;
 }
+
+export function validateBlockNumber(value: string | number | boolean | object) : boolean{
+  // Reject if not a number
+  if(typeof value !== 'number'){
```
Comment (AryanGodara) [inline src/utils/validations.ts:48]: Since we're only concerned with `number` type, there's no actual need for passing specific types for `value` argument.
Try this approach:
```
export function validateBlockNumber(value: any): boolean {
  if (typeof value === 'number') {
    return !isNaN(value) && value >= 0;
  }

  return false;
}
```

File: src/utils/validations.ts [hunk 2]
```diff
@@ -42,3 +42,16 @@ export function validateBlockHash(blockHash: string): boolean {
 
   return true;
 }
+
+export function validateBlockNumber(value: string | number | boolean | object) : boolean{
```
Comment (AryanGodara) [inline src/utils/validations.ts:46]: Since we're only concerned with `number` type, there's no actual need for passing specific types for `value` argument.
Try this approach:
```
export function validateBlockNumber(value: any): boolean {
  if (typeof value === 'number') {
    return !isNaN(value) && value >= 0;
  }

  return false;
}
```

File: src/rpc/calls/getStorageAt.ts [hunk 3]
```diff
@@ -31,15 +31,15 @@ export async function getStorageAtHandler(
   }
 
   const snAddress = await getSnAddressFromEthAddress(ethAddress)
-  const starknet_params = {
```
Comment (AryanGodara) [inline src/rpc/calls/getStorageAt.ts:34]: We can perhaps leave fixing naming conventions for a dedicated PR later (:

File: tests/rpc/calls/getBlockTransactionCountByNumber.test.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { getBlockTransactionCountByNumberHandler } from "../../../src/rpc/calls/getBlockTransactionCountByNumber";
+import { RPCResponse } from "../../../src/types/types";
+
+
+describe('Test getBlockTransactionCountByNumber request testnet', () => {
```
Comment (AryanGodara) [inline tests/rpc/calls/getBlockTransactionCountByNumber.test.ts:5]: Add some failing tests, like
1. InvalidBlockNumber
2. Non existing block
3. invalid/missing parameters

File: src/rpc/calls/getBlockTransactionCountByNumber.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    params: [
+      {
+        block_number: blockNumber as number,
+      },
+    ],
+    id: request.id,
+  })
+
+  if (
+    typeof response === 'string' ||
```
Comment (AryanGodara) [inline src/rpc/calls/getBlockTransactionCountByNumber.ts:44]:
```
if (!response || typeof response === 'string')
```
`!response` is enough to account for both undefined and null ;)

File: src/utils/validations.ts [hunk 6]
```diff
@@ -42,3 +42,16 @@ export function validateBlockHash(blockHash: string): boolean {
...
+  if(typeof value !== 'number'){
+    return false;
+  }
+  if(value < 0){
+    return false;
+  }
+  return true;
+}
+
+
```
Comment (AryanGodara) [inline src/utils/validations.ts:57]: remove the extra empty lines (lint the code, in general)

File: tests/utils/validations.test.ts [hunk 7]
```diff
@@ -72,4 +73,12 @@ describe('Test Address validations', () => {
     const result = validateSnAddress('')
     expect(result).toBe(false)
   })
+  it('return true if block number is a valid number', async () => {
+    const result = validateBlockNumber(123)
+    expect(result).toBe(true)
```
Comment (AryanGodara) [inline tests/utils/validations.test.ts:78]: Add tests for passing invalid (non-numeric) and null/undefined types, since your function handles that logic ;)

**Review by AryanGodara — CHANGES_REQUESTED** [review]
Awesome work here :)
I've suggested a few minor changes

**Comment by AryanGodara** [comment]
> Closes #66

Write `closes #676` so github can detect you're working on that issue ;)

**Comment by ermvrs** [comment]
Please resolve conversations and fix Node CI error.

**Comment by TropicalDog17** [comment]
@ermvrs done sir, please check it out

**Comment by ermvrs** [comment]
please resolve conflicts

**Comment by TropicalDog17** [comment]
> please resolve conflicts

done sir @ermvrs
